src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2433,6 +2424,36 @@ public long GetBitLength()
             return bitLength - 1;
         }
 
+        private static double ConvertToDouble(uint[] bits,
+            int length,
+            int sign)
+        {
```

Comment:
It might be better to try and normalize this to be the same as the already known "good" logic used in our `double` parsing/formatting logic:
<https://source.dot.net/#System.Private.CoreLib/Number.Parsing.cs,9927df209c6039f5,references> and <https://source.dot.net/#System.Private.CoreLib/Number.NumberToFloatingPointBits.cs,3b99f12cbc14a09d>

The overall logic is a bit more complex since it needs to handle non integral numbers; but it also correctly handles rounding and various edge cases.

Reply:
@tannergooding, thatk you for your comment, I had not know about it. I agree, that a widely used method is better than a new one, but I also believe that the provided methods contain more logic than necessary and may work little slower.
I'll try to understand provided methods and apply some one for remove my method.

Reply:
@tannergooding , I can't understand how to use  https://source.dot.net/#System.Private.CoreLib/Number.Parsing.cs,9927df209c6039f5,references , because I can not add reference to [System.Private.CoreLib](https://github.com/dotnet/runtime/tree/main/src/coreclr/System.Private.CoreLib) .
Can you provide little example about reusing [Number.NumberToDouble](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs#L2130)?